x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Verifier.java

```diff
@@ -407,13 +418,17 @@ private static boolean checkGroupByHavingHasOnlyAggs(Expression e, Node<?> sourc
 
             // unwrap function to find the base
             for (Expression arg : sf.arguments()) {
-                arg.collectFirstChildren(c -> checkGroupByHavingHasOnlyAggs(c, source, missing, functions));
+                arg.collectFirstChildren(c -> checkGroupByHavingHasOnlyAggs(c, missing, unsupported, functions));
             }
             return true;
 
         } else if (e instanceof Score) {
-            // Score can't be used for having
-            missing.put(e, source);
+            // Score can't be used in having
+            unsupported.add(e);
+            return true;
+        } else if (e instanceof TopHits) {
```

Comment:
I believe this will not catch `MAX(keyword)` pass by.

Reply:
You're right as the Verifier is checking before the optimisations.